src/platform/silabs/platformAbstraction/GsdkSpam.cpp

```diff
@@ -74,6 +82,32 @@ CHIP_ERROR SilabsPlatform::ToggleLed(uint8_t led)
 }
 #endif // ENABLE_WSTK_LEDS
 
+void SilabsPlatform::StartScheduler()
+{
+    sl_system_kernel_start();
+}
+
+#ifdef SL_CATALOG_SIMPLE_BUTTON_PRESENT
+extern "C" {
+void sl_button_on_change(const sl_button_t * handle)
+{
+    if (Silabs::GetPlatform().mButtonCallback == nullptr)
+    {
+        return;
+    }
+
+    for (uint8_t i = 0; i < SL_SIMPLE_BUTTON_COUNT; i++)
+    {
+        if (SL_SIMPLE_BUTTON_INSTANCE(i) == handle)
+        {
+            Silabs::GetPlatform().mButtonCallback(i, sl_button_get_state(handle));
```

Comment:
should we check if there is a `mButtonCallback` registered

Reply:
It's done above with the nullptr check. Callback is initialized at nullptr, so if nothing was registered we shouldn't get here.